moveit_simple_controller_manager/include/moveit_simple_controller_manager/multi_dof_follow_joint_trajectory_handle.h

```diff
@@ -0,0 +1,135 @@
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the Willow Garage nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Michael Ferguson, Ioan Sucan, E. Gil Jones, Harsh Pandya */
+
+#ifndef MOVEIT_PLUGINS_MULTI_DOF_FOLLOW_JOINT_CONTROLLER_HANDLE
+#define MOVEIT_PLUGINS_MULTI_DOF_FOLLOW_JOINT_CONTROLLER_HANDLE
+
+#include <moveit_simple_controller_manager/action_based_controller_handle.h>
+#include <control_msgs/MultiDOFFollowJointTrajectoryAction.h>
+
+namespace moveit_simple_controller_manager
+{
+
+/*
+ * This is generally used for multi-dof arm joints, but could also be used for multi-dof robot base,
+ *   or anything using a control_mgs/MultiDOFFollowJointTrajectoryAction.
+ */
+class MultiDOFFollowJointTrajectoryControllerHandle : public ActionBasedControllerHandle<control_msgs::MultiDOFFollowJointTrajectoryAction>
+{
+public:
+
+  MultiDOFFollowJointTrajectoryControllerHandle(const std::string &name, const std::string &action_ns) :
+    ActionBasedControllerHandle<control_msgs::MultiDOFFollowJointTrajectoryAction>(name, action_ns)
+  {
+  }
+
+  virtual bool sendTrajectory(const moveit_msgs::RobotTrajectory &trajectory)
+  {
+    ROS_DEBUG_STREAM("MultiDOFFollowJointTrajectoryController: new trajectory to " << name_);
+
+    if (!controller_action_client_)
+      return false;
+
+      /*
+       * This should never happen as the moveit_simple_controller_manager loads
+       * MOVEIT_PLUGINS_FOLLOW_TRAJECTORY_CONTROLLER_HANDLE for single-dof joints
+       */
+    if (!trajectory.joint_trajectory.points.empty())
```

Comment:
again, this comment is misleading -- if the controllers.yaml is wrong, this will hit.